### ISSUE TITLE
Support pure Keras 3 keras_mdn_layer

### ISSUE THREAD

**Issue by cpmpercussion** [cpmpercussion/impsy#64]
A pure keras 3 version of the `keras_mdn_layer` is ready and available for testing over at <https://github.com/cpmpercussion/keras-mdn-layer/releases/tag/v0.7.0-alpha>

The goals here are:

1. change the dependency from keras_mdn_layer  v0.6.0 to v0.7.0-alpha and eliminate tensorflow_probability, _eventually_ I want to move to tf 2.20.0 and python 3.13, but in the interest of moving slowly, testing with python 3.12 and tf 2.18.1 would be sensible.
2. update code as necessary to ensure that tests pass paying particular attention to the tflite interpreter as this is most important runtime inference model (due to significant documented speedups against the Keras model).
3. ensure that existing tflite models still work with the new layer and interpreters, e.g., the bundled config for Roland S-1 `poetry run ./start_impsy.py run -c configs/roland-s-1.toml` needs to be operational for continued gigs.
4. carefully test training procedures with some real data to see how the new layer loss functions are working.
